settings.md
layout: page
title: "SETTINGS"
permalink: /settings/

Pre-configure your settings before download
Testing


preconfigure.md
layout: page
title: "SETTINGS"
permalink: /preconfigure/

Pre-configure your settings before download
Testing


CHANGE_LOG.md
# Change Log

**Version W34-HC-IMJD-2.0.0**

1. Changes for WeeWX 4.0 new extended database schema
2. Protection code for empty strings
3. Added energy to lightning module display
4. CSS additions in preparation for future planned air quality module options
5. Updated settings.php to next version code **W34-HC-IMJD-2.0.0**
6. Added lightning chart links to index.php
7. Code cleanup lightning variables & css tweaks for lightning
8. Added lightning monthly chart
9. Add missing y axis label for lightning
10. Additional lightning label changes
11. Readme up issue
12. Removed duplicated documents in www folder
13. Adjust location of miles widget
14. Reduce race condition on update section
15. wind_gust_60min missing and wind conversions in the wrong file
16. Cloud height display reversed
17. CSS adjustment for airqualitymodule.php
18. Corrected issues with RT windrose chart
19. Further air quality module CSS adjustments
20. Air Quality Module updated
21. AQI info pop up added (aqipopup.php, aqi_info.php)
22. World AQI Top Row Module added (worldaqitop.php)
23. UK AQI Top Row added (daqitop.php)
24. Added cloudbase data to archivedata.php.tmpl
25. w34CombineData.php now uses WeeWX cloud base data 26 Fixed cloud base unit conversion in currentconditions.w34.php
26. Use lasttime for lightning strike
27. Created airqualitydata.php.tmpl
28. Added [[[AIRQUALITYDATA]]] to Weather34 skin.conf
29. Dual windrose plots on one iframe
30. Day plots for multiple plots per page
31. Cloudbase calculation works on US units database also
32. Month/Year Windrose Gust use GustSpeed from db
33. Windrose Gust tweaks and using 10min gust for display
34. Fix chk_lightning_count option
35. Remove unused config entry
36. Weewx IP in realtime.txt for day plots
37. Remove weewxbinpath from plots_config.js
38. Change to make lightning time ago correct
39. Round 2 Day Plots communication
40. Use 10min dir avg and 10min gust avg for rose
41. Config option to disable day plots
42. New Cloud Cover percent display
43. New Cloud Cover Chart
44. Misc changes for Cloud Cover and chmod for sat images
45. Added cloud cover unit to settings.php
46. Renamed RemoteWebServer and realtime.txt
47. Added text to cloud cover chart
48. Added additional hidden fields for new hidden variables to ensure they are not removed from settings1.php when settings page is saved.
49. Remove scroll bar from temp almanac
50. Corrected path in Weather34CloudCover stanza in the install config files
51. Added % cloud cover chart link to current conditions module
52. Rsync w34realtime.txt to remote web servers
53. Rsync of json directory is now possible
54. Remove duplicate filename config entries in W34 sections
55. Added green, yellow, orange and red advisory alert triangles to shared.php
56. Inserted space in cloud cover link index.php
57. Fixed white plot background for cloud cover chart
58. Buttons styles
59. W34 WebServer always enabled remove config section from weewx conf
60. Send webserver ip address on index.php page load
61. currentconditionsw34.php adjusted layout
62. Support multiple remote web servers for one weewx machine
63. Auto theme for dark and light theme based on sunrise & sunset
64. Added delete of config entries to w34 installer
65. Added advisory_extended.php
66. Added alert fields to easyW34skinSetup.php
67. $lat and $lon added to auto theme index.php
68. settings1.default.php additional variables for Meteoalarm added
69. Additional updates to easyW34skinSetup.php for Meteoalarm filelds
70. Hayes wind speed chart and range selector for radial charts
71. index.php edited module links
72. Protection code around lightningSum_json
73. Protection code for degrees to compass value
74. Cloud Cover Radial chart added
75. readme.md Updated screen shots
76. Updated radial chart screen shot
77. Added polar chart for tempallplot
78. Use either weewx 3.x or 4.x logging
79. Allow polar charts to use different dates using input selector
80. User is now prompted with the Python Version the install will be run
81. Re-worked cloud cover
82. Added night and day icon selection to cloud conditions
83. easyW34skinSetup.php MB platform not required
84. Added $theme1 and $al_ip to settings1.default.php
85. Added missing cloud cover variable to w34CombinedData.php
86. Current conditions module Icons and descriptions now match
87. Addition of oktas to cloud coverage in current conditions module
88. Readme and installation instructions updated
89. Updated weather34.py, change for time out in P3
90. Added w34_reports to w34Highcharts folder and updated plot_config.js accordingly
91. Rolled back 89. to previous state as it caused chart click-through to hang on final click
92. weather34.py corrected hard coded path
93. w34CombinedData.php try catch around loading archivedata.php 
94. Fix for Highcharts update.
95. Corrected font weight WU large forecast in light theme


www/settings.php
<?php 
include('settings1.php');error_reporting(0); 
	####################################################################################################
	#	HOME WEATHER STATION TEMPLATE SETUP please set up and check thoroughly                         #
	#	CREATED FOR HOMEWEATHERSTATION TEMPLATE at https://weather34.com/homeweatherstation/index.html # 
	# 	   FILE= SETTINGS.PHP                                                                          #
	# 	                                                                                               #
	# 	WEATHER34 HOMEWEATHER METEOBRIDGE MYSQL                                                        #
	#   https://www.weather34.com 	                                                                   #
	####################################################################################################


###########################################################################
# THE SETIINGS BELOW MUST BE LEFT UNTOUCHED UNLESS YOU REALLY NEED  #######
# TO MAKE THE CHANGES HERE TO MAKE IT WORK ON YOUR SERVER// ###############
# TAKE THE TIME TO STUDY THEM DONT TAKE IT FOR GRANTED      ############### 
# USE THE EASY SETUP PANEL TO MAKE CHANGES IN THE TEMPLATE  ###############
###########################################################################

$TZconf = $TZ;// PLEASE LEAVE it is fixed 
$api2= 'xxx'; // optional Weather Underground api key developer key '1234567890'
$rise_zenith = 90+ 40/60; // try 50/60 or something/60 until it matches correctly to your sunrise .this allows you to fine tune the sunrise
$set_zenith = 90+ 36/60;  // try 50/60 or something/60 until it matches correctly to your sunset .this allows you to fine tune the sunset
$forecastlocation = $stationlocation; // 
$version            = $livedataFormat; // template version and type of source: Clientraw, MeteoBridge, Cumulus, etc (for display only)
$emailform          = $email; // PLEASE LEAVE FIXED 
$timeFormatShort    = 'H:i'; // time format to use when showing times ('g:i a' for '5:34 pm'; 'H:i' for '09:28')
$showFeelsLike      = true; // whether to always show either the heat index (when temp > 80F/27C) or real feel (when temp between 50F/10C and 80F/27C) even when no concern
$lightLeft          = true; // shows amount of light/darkness left rather than total amount per day in the moon display
$theme              = $theme1;
$charttheme         = $theme;

####################################################################################################
// Refresh Data Main Page  //																	   #
// Automatic refresh times (in seconds) of each panel on the main dashboard						   #
####################################################################################################
$copyYear = 2015;
$curYear = date('Y');
$copyrightcredit='&copy; weather34.com original CSS/SVG/PHP '.$copyYear . (($copyYear != $curYear) ? '-' . $curYear : 'Copyright');
$indoorRefresh      = 600; // 10 minutes
$advisoryRefresh    = 60; // 60 seconds
$tempRefresh        = 60; // 60 seconds
$skyRefresh         = 600; // 10 minutes
$windSpeedRefresh   = 04; // 4 seconds
$baroRefresh        = 300; // 5 minutes
$windDirRefresh     = 04; // 4 seconds
$solarRefresh       = 600; // 10 minutes
$moonRefresh        = 1800; // 30 minutes 
$rainRefresh        = 60; // 60 seconds 
$p12Refresh         = 600; // 10 minutes        Position 12 refresh (if not webcam) 
$p13Refresh         = 600; // 10 minutes        Last Postion refresh (if not webcam)
$moonphaseRefresh   = 600; // 10 minutes
$eqRefresh          = 600; // 10 minutes 
$forecastRefresh    = 900; // 15 minutes 
$notificationcenterRefresh    = 300; // 5 minutes set refresh time for notification of earthquakes default 5 minutes
$wuupdate           = 1800; // 30 minutes 
$wfupdate           = 60; // weatherflow api 60 seconds
$camRefresh         = 60; // 1 minute refresh on camera popup
// Thresholds for warnings or notifications
$notifyDistEQ       = 300; // miles if $windunit is mph otherwise km
$notifyEQMag        = 5; // set this to minimum magnitude of earthquake for notifications, typically 5 or 6, or you can set to 12 to switch off if not required
$notifyWindGust     = 22; // knots; 22 knots, which is 40.7 km/h, 35 = 40.2 mph, 40 = 20.5  m/s
$notifyHeatIndex    = 30; // degrees C, default is 30 celsius which is 86 degrees F 
$notifyWindchill    = 3; // degrees C, default is 10 celsius which is 50 degrees F
$notifyFeelslike    = 3; // degrees C, default is 10 celsius which is 50 degrees F
$notifyDewpointcold  = 3; // degrees C, default is 8 celsius 46.4 degrees F OR WHATEVER MAKES YOU FEEL COLDER
$notifyRainRate     = 7.62; // millimeters, default is 10 which is 0.4 in"
$notifyDewPoint     = 21; // dewpoint comfortable zone alert 22 degrees C or 71.6 degrees F OR WHATEVER MAKES YOU FEEL UNCOMFORTABLY HOT
$notifyFlooding     = 25.4; // dewpoint comfortable zone alert 22 degrees C or 71.6 degrees F OR WHATEVER MAKES YOU FEEL UNCOMFORTABLY HOT

####################################################################################################
// Probably won't have to change anything past this line IF SO BE VER VERY CAREFUL!!!!!!		   #
####################################################################################################
$scriptcredits     = "Original CSS/SVG ICONS/PHP scripts by <a href='https://weather34.com' title='weather34.com' target='_blank'>weather34.com 2015 - ".date('Y').""; // for modules
$creditsEnabled     = "true"; // for chart pages only
$creditsmysql       = "Weather Data Recorded from $stationName Database"; //for chart pages only
$wucredits          = "Data provided by Weather Underground"; // for chart pages only
$creditschart 		= "CanvasJs.com v2.3.1 GA (CC BY-NC 3.0) Non-Commercial-Version";
$creditsURL         = ""; // for chart pages only
$credits            = "Data Supplied via Weather Underground"; // for chart pages only
$templateinfo       = ''; // template information page
$templateversion    = 'W34-HC-<maxblue>IMJD</maxblue>-2.0.0';
$software    = 'WeeWX <span>Hardware</span> Users';$designedfor='<br>For WeeWX Users';$chartversion='(WUDATACHARTS)';$somethinggoeshere ='d4586dec-e7a2-47ae-99b6-25527b2563c9';$chartversionmysql  =  '(DATACHARTS WeeWX Database Version)';
if (array_key_exists('theme', $_GET) && ($_GET['theme'] == 'dark' || $_GET['theme'] == 'light')) {
  SetCookie('theme', $_GET['theme'], time() + 15552000);
  $theme = $_GET['theme'];
} else if (array_key_exists('theme', $_COOKIE) && ($_COOKIE['theme'] == 'dark' || $_COOKIE['theme'] == 'light')) {
  $theme = $_COOKIE['theme'];
}
$units = "";
if (array_key_exists('units', $_COOKIE)) {
  $units = $_COOKIE['units'];
}
ini_set('session.use_cookies', '0');
if (array_key_exists('units', $_GET) && ($_GET['units'] == 'default' || $_GET['units'] == '')) {
  SetCookie('units', $_GET['units'], time() - 86400); //86400 = 1 day, negative time erases cookie
  $units = '';
} else if (array_key_exists('units', $_GET) && ($_GET['units'] == 'us' || $_GET['units'] == 'uk' || $_GET['units'] == 'metric' || $_GET['units'] == 'scandinavia')) {
  SetCookie('units', $_GET['units'], time() + 15552000);
  $units = $_GET['units'];
}
if ($units == 'uk') {
  $cloudcoverunit   = '%';	
  $windunit         = 'mph';
  $tempunit         = 'C';
  $rainunit         = 'mm';
  $pressureunit     = "hPa";
  $windconv         = "0.621371";
  $rainfallconv     = '10';
  $pressureinterval = "0.5";
  $rainfallconvmm   = '10';
} else if ($units == 'scandinavia') {
  $windunit         = 'm/s';
  $tempunit         = 'C';
  $rainunit         = 'mm';
  $pressureunit     = "hPa";
  $windconv         = "0.277778";
  $rainfallconv     = '10';
  $pressureinterval = "0.5";
  $rainfallconvmm   = '10';
} else if ($units == 'metric') {
  $windunit         = 'km/h';
  $tempunit         = 'C';
  $rainunit         = 'mm';
  $pressureunit     = "hPa";
  $windconv         = "1";
  $rainfallconv     = '10';
  $pressureinterval = "0.5";
  $rainfallconvmm   = '10';
} else if ($units == 'us') {
  $windunit         = 'mph';
  $tempunit         = 'F';
  $rainunit         = 'in';
  $pressureunit     = "inHg";
  $windconv         = "1";
  $rainfallconv     = '1';
  $pressureinterval = "0.5";
  $rainfallconvmm   = '1';
}
?>
